plugins/catalog/src/components/ComponentsCard/HasComponentsCard.tsx

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ComponentEntity, RELATION_HAS_PART } from '@backstage/catalog-model';
+import {
+  CodeSnippet,
+  InfoCard,
+  Link,
+  Progress,
+  WarningPanel,
+} from '@backstage/core';
+import {
+  EntityTable,
+  useEntity,
+  useRelatedEntities,
+} from '@backstage/plugin-catalog-react';
+import React, { PropsWithChildren } from 'react';
+
+const ComponentsCard = ({
+  children,
+  variant = 'gridItem',
+}: PropsWithChildren<{ variant?: string }>) => {
+  return (
+    <InfoCard variant={variant} title="Components">
+      {children}
+    </InfoCard>
+  );
+};
+
+type Props = {
+  variant?: string;
+};
+
+export const HasComponentsCard = ({ variant = 'gridItem' }: Props) => {
+  const { entity } = useEntity();
+  const { entities, loading, error } = useRelatedEntities(entity, {
+    type: RELATION_HAS_PART,
+    kind: 'Component',
+  });
+
+  if (loading) {
+    return (
+      <ComponentsCard variant={variant}>
```

Comment:
yeah i keep thinking the same here as above - why doesn't this one just render `<InfoCard variant={variant}>` and then move all the useEntity and stuff into an inner contents component?

Reply:
I think this is just a left over from a time where I had to repeat more code. I will just use the InfoCard directly.